curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641da5abaac81844a54adb03.md

```diff
@@ -7,40 +7,78 @@ dashedName: step-8
 
 # --description--
 
-Now you'll create a function to populate the UI with the author data. You will call this function inside the second `.then()` method.
+Now, it is time to create the function that will be responsible for displaying the authors on the page. 
 
-Create an empty arrow function named `displayAuthors` that takes `authors` as a parameter.
+Start by creating a function called `displayAuthors` that takes an `authors` parameter. Your function should then loop through the `authors` array and add a card for each author to the `authorContainer` `div`. 
 
-# --hints--
-
-You should use `const` to create an empty function named `displayAuthors`.
+The card should have the following structure:
 
-```js
-assert.match(code, /const\s+displayAuthors\s*=\s*/)
+```html
+<div id="${index}" class="user-card">
+  <h2 class="author-name">${author}</h2>
+</div>
 ```
 
-`displayAuthors` should be a function.
+**Tips:**
 
-```js
-assert.isFunction(displayAuthors)
-```
+- You can use any type of loop to iterate over the `authors` array. Ex. `for`, `forEach`, `map`, etc.
+- You should use the `innerHTML` property to add the card to the `authorContainer` inside the loop. You will also need to use template literal syntax to add the markup.
+
+# --hints--
 
-Your `displayAuthors` function should use an arrow syntax.
+`displayAuthors` should be a function.
 
 ```js
-assert.match(code, /const\s+displayAuthors\s*=\s*(\(.*\)|[^\s()]+)\s*=>\s*/)
+assert.isFunction(displayAuthors)
 ```
 
 Your `displayAuthors` function should take an `authors` parameter.
 
 ```js
-assert.match(code, /const\s+displayAuthors\s*=\s*\(?\s*authors/)
+const parameters = __helpers.getFunctionParams(displayAuthors.toString());
+assert.deepEqual(parameters[0].name,"authors"); 
 ```
 
-Your `displayAuthors` function  should be empty.
+When the `displayAuthors` function is called with an author's array, it should add a card for each author to the `authorContainer` `div`.
 
 ```js
-assert.match(code, /const\s+displayAuthors\s*=\s*(\(\s*authors\s*\)|authors)\s*=>\s*\{\n?\s*?\}\s*;?/)
+const exampleAuthors = [
+[
+  {
+    "author": "Quincy Larson",
+  },
+  {
+    "author": "Jessica Wilkins",
+  },
+],
+[ 
+{
+    "author": "Naomi Carrigan",
+  },
+  {
+    "author": "Beau Carnes",
+  },
+  {
+    "author": "David Clinton",
+  }
+],
+];
+
+for (const authorSet of exampleAuthors) {
+  displayAuthors(authorSet);
+
+  const authorCards = document.querySelectorAll("#author-container div");
+  const authorTitles = document.querySelectorAll("#author-container div h2");
+  assert.deepEqual(authorCards.length,authorTitles.length); 
+  for(let i = 0; i < authorCards.length; i++)
+  {
+    assert.deepEqual(authorCards[i].className,"user-card");
+    assert.deepEqual(authorCards[i].id,i.toString());
+    assert.deepEqual(authorTitles[i].className,"author-name");
+    assert.deepEqual(authorTitles[i].innerText,authorSet[i].author);
+  }
```

Comment:
fix formatting here. It is currently giving me C vibes instead of JS vibes

Reply:
I don't quite understand what you mean by C vibes but I would be happy to fix it.